tests/system/test_client.py

```diff
@@ -335,6 +349,46 @@ def test_create_table(self):
         self.assertTrue(_table_exists(table))
         self.assertEqual(table.table_id, table_id)
 
+    def test_create_tables_in_case_insensitive_dataset(self):
+        ci_dataset = self.temp_dataset(
+            _make_dataset_id("create_table"), is_case_insensitive=True
+        )
+        table_arg = Table(ci_dataset.table("test_table2"), schema=SCHEMA)
+        tablemc_arg = Table(ci_dataset.table("Test_taBLe2"))
+
+        table = helpers.retry_403(Config.CLIENT.create_table)(table_arg)
+        self.to_delete.insert(0, table)
+
+        self.assertTrue(_table_exists(table_arg))
+        self.assertTrue(_table_exists(tablemc_arg))
+        self.assertIs(ci_dataset.is_case_insensitive, True)
+
+    def test_create_tables_in_case_sensitive_dataset(self):
+        ci_dataset = self.temp_dataset(
+            _make_dataset_id("create_table"), is_case_insensitive=False
+        )
+        table_arg = Table(ci_dataset.table("test_table3"), schema=SCHEMA)
+        tablemc_arg = Table(ci_dataset.table("Test_taBLe3"))
```

Comment:
another nit: what does `mc` stand for in the variable `tablemc_arg`? Could we use something easier to understand?

Reply:
`mc` = mixed case; added a comment about that